src/_pytest/_io/saferepr.py

```diff
@@ -33,28 +53,20 @@ def repr_instance(self, x, level):
         return self._callhelper(repr, x)
 
     def _callhelper(self, call, x, *args):
-        try:
-            # Try the vanilla repr and make sure that the result is a string
-            s = call(x, *args)
-        except Exception:
-            cls, e, tb = sys.exc_info()
```

Comment:
I'd keep the call here, and handle the exception via `_safe_format_exc`.

Reply:
Could you please explain it a bit? Do you mean something like
```
try:
    # Try the vanilla repr and make sure that the result is a string
    s = call(x, *args)
except Exception as exc:
    return _safe_format_exception(exc, x)
else:
    ...
```
and use `_safe_format_exception` here and in `_call_and_format_exception`?

Reply:
Oh, sorry - forgot to remove that comment, after noticing that it is used twice here now (was confused by the name also).